qadence/ml_tools/constructors.py

```diff
@@ -733,42 +746,68 @@ def build_qnn_from_configs(
         fm_config (FeatureMapConfig): Feature map configuration.
         ansatz_config (AnsatzConfig): Ansatz configuration.
         observable_config (ObservableConfig): Observable configuration.
+        backend (BackendName): The chosen quantum backend.
+        diff_mode (DiffMode): The differentiation engine to use. Choices are
+            'gpsr' or 'ad'.
+        measurement (Measurements): Optional measurement protocol. If None,
+            use exact expectation value with a statevector simulator.
+        noise (Noise): A noise model to use.
+        configuration (BackendConfiguration | dict): Optional backend configuration.
+        input_diff_mode (InputDiffMode): The differentiation mode for the input tensor.
 
     Returns:
         QNN: A QNN model.
     """
-    fm_blocks = create_fm_blocks(register=register, config=fm_config)
-    full_fm = _interleave_ansatz_in_fm(
-        register=register,
-        fm_blocks=fm_blocks,
-        ansatz_config=ansatz_config,
-    )
 
-    ansatz = create_ansatz(register=register, config=ansatz_config)
+    if observable_config is None:
```

Comment:
Why making `observable_config` optional if you check existence later ? Wouldn't it be simpler to keep it as positional argument ?

Reply:
@Roland-djee Thanks for the review!

The reason why the these configs default to `None` and then checked, although a bit ugly, is there to maintain the FM, ansatz, observable order.

And some users have asked to not have to include FM in their QNNs.

Reply:
Yeah I got that point. But I guess we can get away without making them optional, as discussed below.